### PR TITLE
Changes for sqlalchemy 2.0.0 compatibility

### PR DESCRIPTION
**Pull request by jrast**
This is a PR to provide the upcomming SQLAlchemy Release 2.0. There is no hurry to integrate this as fast as possible, currently (Sept. 2022) SQLAlchemy Version 2.0 is still in beta. However: Better be prepared!

The changes are made according to the [Version 2.0 Migration Guide](https://docs.sqlalchemy.org/en/20/changelog/migration_20.html)

### CODE REVIEW

**Comment by jrast** [comment]
Note: To check compatibility with SQLAlchemy Version 2.0, the deprecation warnings can be enabled during test execution:
````powershell
# Powershell

# Enable warnings
$env:SQLALCHEMY_WARN_20=1; pytest -W always .\tests

# Treat warnings as errors. See note below
$env:SQLALCHEMY_WARN_20=1; pytest -W error::sqlalchemy.exc.SADeprecationWarning .\tests
````
If the warnings are treated as errors, make sure to only enable sqlalchemy warnings as errors. As long as flask-sqlalchemy version 3.0.0 is not released, the extension will raise deprecation warnings as deprecated methods of flask are used.

**Comment by jwag956** [comment]
Yea - not too keen on the second commit - looking at the documentation - model.query isn't going away anytime soon and isn't even deprecated. Your first commit looks great.  Thanks

**Comment by jrast** [comment]
The only advantage I see with the new style is that we become less dependent on flask-sqlalchemy, since we no longer have to rely on the `Model.query` property. This might make it easier to use the package without flask-sqlalchemy, but as I just found out, the query property can be added to a custom base quite easily (as shown in [the docs](https://flask-security-too.readthedocs.io/en/stable/quickstart.html#id4)).

File: tests/test_datastore.py [hunk 1]
```diff
@@ -657,16 +658,17 @@ class User(db.Model, fsqla.FsUserMixin):
...
+        meta_data.reflect(db.engine)
         role_table = meta_data.tables["role"]
 
         # Start by manually creating a role in the 4.1.x style
         stmt = insert(role_table).values(
             name="r1", description="r1 v41", permissions="read,write"
         )
         with db.engine.connect() as conn:
-            conn.execute(stmt)
+            with conn.begin():
```
Comment (jwag956) [inline tests/test_datastore.py:670]: can't this be just with db.engine.connect().begin(): xxx?

File: flask_security/datastore.py [hunk 2]
```diff
@@ -15,6 +15,11 @@
 
 from .utils import config_value as cv
 
+try:
```
Comment (jwag956) [inline flask_security/datastore.py:18]: move this into the class so don't need a try/except?

**Review by jwag956 — COMMENTED** [review]
Looking further - I do like the idea that the SQLAlchemySessionUserDatastore not actually relying on Flask-SQLAlchemy constructs....

**Comment by jwag956** [comment]
How about you revert the second commit - the first one is easy, simple, and a nice addition.
I believe that SQLAlchemySessionUserDatastore already shows how to use FS w/o FlaskSqlAlchemy.

The current code uses model.query which is really base.db_session.query_property - which I think is pretty much inline with 2.0? Am I missing something?

File: tests/test_datastore.py (hunk 1 above)
Comment (jrast) [reply tests/test_datastore.py:670]: I don't think that it's possible to these two context managers in a single call. But I can try if you want.

**Comment by jrast** [comment]
I reverted the second commit. 
You are right, `model.query` is just a `query_property`. As the `query` interface is considered legacy in SQLAlchemy 2.0 there might be a good reason to switch to the new style in the far future, but currently it would be more like a "premature optimisation" which is not worth it.

**Comment by jwag956** [comment]
Thanks - if you rebase - the failing tests should be fixed - minor change due to Flask-SQLAlchemy 3.0.

Thanks for your time.

**Comment by jrast** [comment]
Ok, rebased.

**Review by jwag956 — APPROVED** [review]

**Pull request merged**